arelle/FileSource.py

```diff
@@ -655,6 +655,19 @@ def file(
         else:
             return openXmlFileStream(self.cntlr, filepath, stripDeclaration)
 
+    def getBytesSize(self) -> int | None:
+        """
+        Get the size of the zip file in bytes.
+        :return: Size of the zip file in bytes, or None if not applicable.
+        """
+        if isinstance(self.basefile, str) and os.path.isfile(self.basefile):
+            return os.path.getsize(self.basefile)
+        if isinstance(self.fs, zipfile.ZipFile) and self.fs.fp is not None:
```

Comment:
If we're trying to be conservative, should it be `getattr(self.fs, 'fp')`, or do we want it to fail if the private `fp` attribute is renamed?  In any case, I wonder if it's worth a comment explaining that we're using a private field.